### PR TITLE
Update README with helm info

### PR DESCRIPTION
**Pull request by tiswanso**
Show how to use helm to generate a manifest and explain options.  Indicate currently available `values.yaml` files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+
+   1. helm chart params
+   
+      | Option | Values | Description |
+      |--------|--------|-------------|
+      | hub | | The container registry to pull the `install-cni` container from. |
+      | tag | | The container tag to use to pull the `install-cni` container. |
+      | logLevel | panic, fatal, error, warn, info, debug | Logging level for CNI binary |
+      | excludeNamespaces | `[]string` | list of namespaces to exclude from Istio pod check |
+      | cniBinDir | | Use this to set to the environment's `--cni-bin-dir` setting (kubelet param) |
```
Comment (baodongli) [inline README.md]: can you make it something like: the same as kubelet's --cni-bin-dir setting

File: README.md [hunk 2]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+      | cniConfDir | | Use this to set to the environment's `--cni-conf-dir` setting (kubelet param) |
```
Comment (baodongli) [inline README.md]: ditto

File: README.md [hunk 3]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+      | cniConfFileName | | Use this to set to the environment's CNI config filename in the `cni-conf-dir`.  Leave unset to auto-find the first file in the `cni-conf-dir`. |
```
Comment (baodongli) [inline README.md]: kubelet doesn't support this arg. Please indicate that it's not used for deployment

File: README.md (hunk 1 above)
Comment (tiswanso) [reply README.md]: Sure, I'll fix the wording here

File: README.md [hunk 4]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+   
+      | Environment | helm values |
+      |-------------|-------------|
+      | default kubeadm | [values.yaml](deployments/kubernetes/install/helm/values.yaml) |
+      | GKE | [values_gke.yaml](deployments/kubernetes/install/helm/values_gke.yaml) |
+
+   1. helm chart params
+   
+      | Option | Values | Description |
+      |--------|--------|-------------|
```
Comment (baodongli) [inline README.md]: would be nice to add default values

Comment (tiswanso) [reply README.md]: sure

**Comment by tiswanso** [comment]
@venilnoronha  agreed.  I can also add the helm package and helm serv instructions in this one at that point.

File: README.md [hunk 5]
```diff
@@ -143,6 +164,44 @@ $ GOOS=linux make docker.push
 
 **NOTE:** Set HUB and TAG per your docker registry.
 
+### Helm
+
+The helm package tarfile can be created via
```
Comment (venilnoronha) [inline README.md]: nit: helm -> Helm, here and at other non-code locations.

File: README.md [hunk 6]
```diff
@@ -143,6 +164,44 @@ $ GOOS=linux make docker.push
...
+
+```
+helm package $GOPATH/src/istio.io/deployments/kubernetes/install/helm/istio-cni
+```
+
+#### Serve Helm Repo
+
+An example for hosting a test repo for the helm istio-cni package:
+1. Create package tarfile with `helm package $GOPATH/src/istio.io/deployments/kubernetes/install/helm/istio-cni`
+1. copy tarfile to dir to serve the repo from
```
Comment (venilnoronha) [inline README.md]: nit: copy -> Copy.

File: README.md [hunk 7]
```diff
@@ -143,6 +164,44 @@ $ GOOS=linux make docker.push
...
+1. Run `helm serve --repo-path <dir where helm tarfile is>`
+   1. The repo URL will be output (`http://127.0.0.1:8879`)
+
+To use this repo via `helm install`:
+1. `helm repo add local_istio http://127.0.0.1:8879`
+1. `helm repo update`
```
Comment (venilnoronha) [inline README.md]: It's preferable to put these longer commands in a code block instead. Also, mark with \```sh wherever needed.

File: README.md [hunk 8]
```diff
@@ -143,6 +164,44 @@ $ GOOS=linux make docker.push
...
+
+At this point the `istio-cni` chart is ready for use by `helm install`.
+
+To make use of the `istio-cni` chart from another chart:
+1. Add the following to the other chart's `requirements.yaml`:
+
+   ```
```
Comment (venilnoronha) [inline README.md]: nit:

<pre>```yaml</pre>

**Comment by john-a-joyce** [comment]
Although not specific to this review I think you (or I could do it) minimize the comparison to other approaches.   IMO that shouldn't be in this README as anyone not intimately involved with Istio and the history of the discussion will be lost.   It's fine to compare to what is currently in Istio as that is a publicly documented approach.

File: README.md [hunk 9]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+1. (Manual Option) Modify [istio-cni.yaml](deployments/kubernetes/install/manifests/istio-cni.yaml)
    1. set `CNI_CONF_NAME` to the filename for your k8s cluster's CNI config file in `/etc/cni/net.d`
    1. set `exclude_namespaces` to include the namespace the Istio control-plane is installed in
    1. set `cni_bin_dir` to your kubernetes install's CNI bin location (the value of kubelet's `--cni-bin-dir`)
       1. default is `/opt/cni/bin`
 
-1. Install `istio-cni`: `kubectl apply -f deployments/kubernetes/install/manifests/istio-cni.yaml`
+1. (Helm Option) Construct a `helm template` or `helm install` command for your Kubernetes environment
+
+   1. `helm template deployments/kubernetes/install/helm/istio-cni --values deployments/kubernetes/install/helm/istio-cni/values.yaml --namespace kube-system --set hub=$HUB --set tag=$TAG > $HOME/istio-cni.yaml`
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: yeah... I agree but I think it depends on the final way it's included in the main istio chart.  For example, in the context of use as a dependency in requirements.yaml of the `istio` chart, the helm command line would be able to use `--set istio-cni.hub=" and `--set istio-cni.tag=` for those.  But @sdake mentioned that it's better to namespace these explicitly under the `istio-cni` key in values.yaml; but that doesn't seem to work right (at least without changing the template).

File: README.md [hunk 10]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
 
 1. Install Istio control-plane
 
-1. Modify [istio-cni.yaml](deployments/kubernetes/install/manifests/istio-cni.yaml)
+1. (Manual Option) Modify [istio-cni.yaml](deployments/kubernetes/install/manifests/istio-cni.yaml)
```
[the remark this answers is not in the document]
Comment (john-a-joyce) [reply README.md]: Locally most of the point numbers look fine to me,  but it Venil comment is valid that it should be clarified that is either option 3 or option 4 and that all the other steps are the same.

[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: sure on nested.  @venilnoronha  I don't follow your "1->3" comment.  You referring to the un-rendered README.md having all 1's?

File: README.md [hunk 11]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
    1. set `exclude_namespaces` to include the namespace the Istio control-plane is installed in
    1. set `cni_bin_dir` to your kubernetes install's CNI bin location (the value of kubelet's `--cni-bin-dir`)
       1. default is `/opt/cni/bin`
 
-1. Install `istio-cni`: `kubectl apply -f deployments/kubernetes/install/manifests/istio-cni.yaml`
+1. (Helm Option) Construct a `helm template` or `helm install` command for your Kubernetes environment
+
+   1. `helm template deployments/kubernetes/install/helm/istio-cni --values deployments/kubernetes/install/helm/istio-cni/values.yaml --namespace kube-system --set hub=$HUB --set tag=$TAG > $HOME/istio-cni.yaml`
+   
+      | Environment | helm values |
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: done

File: README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: done

File: README.md [hunk 12]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+      |-------------|-------------|
+      | default kubeadm | [values.yaml](deployments/kubernetes/install/helm/istio-cni/values.yaml) |
+      | GKE | [values_gke.yaml](deployments/kubernetes/install/helm/istio-cni/values_gke.yaml) |
+
+   1. helm chart params
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: done

File: README.md [hunk 13]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+   
+      | Option | Values | Default | Description |
+      |--------|--------|---------|-------------|
+      | hub | | | The container registry to pull the `install-cni` container from. |
+      | tag | | | The container tag to use to pull the `install-cni` container. |
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: done

File: README.md [hunk 14]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+      | logLevel | panic, fatal, error, warn, info, debug | `warn` | Logging level for CNI binary |
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: done

File: README.md [hunk 15]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+      | excludeNamespaces | `[]string` | `[ istio-system ]` | list of namespaces to exclude from Istio pod check |
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: done

File: README.md [hunk 16]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
+      | cniBinDir | | `/opt/cni/bin` | Must be the same as the environment's `--cni-bin-dir` setting (kubelet param) |
+      | cniConfDir | | `/etc/cni/net.d` | Must be the same as the environment's `--cni-conf-dir` setting (kubelet param) |
+      | cniConfFileName | | None | Leave unset to auto-find the first file in the `cni-conf-dir` (as kubelet does).  Primarily used for testing `install-cni` plugin config.  If set, `install-cni` will inject the plugin config into this file in the `cni-conf-dir` |
+
+1. Install `istio-cni`: `kubectl apply -f $HOME/istio-cni.yaml`
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: done

File: README.md [hunk 17]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
...
 
 1. remove the `initContainers` section from the result of helm template's rendering of
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: done

File: README.md [hunk 18]
```diff
@@ -28,13 +28,34 @@ The following are the steps to install and use the CNI plugin.
 
 1. Install Istio control-plane
```
[the remark this answers is not in the document]
Comment (tiswanso) [reply README.md]: IMO, it's MUCH better to let the markdown rendering take care of the numbering.  I'd prefer to leave this.

**Comment by tiswanso** [comment]
@john-a-joyce  -- IMO, Let's follow-on with cleanup of comparisons and extemporaneous details that are now in the design spec.  We can now refer to the design spec.

File: README.md [hunk 19]
```diff
@@ -28,15 +28,45 @@ The following are the steps to install and use the CNI plugin.
...
-
-1. Install `istio-cni`: `kubectl apply -f deployments/kubernetes/install/manifests/istio-cni.yaml`
-
-1. remove the `initContainers` section from the result of helm template's rendering of
+1. Create Istio CNI installation manifest--either manually or via Helm:
+   1. (Manual Option) Modify [istio-cni.yaml](deployments/kubernetes/install/manifests/istio-cni.yaml)
+      1. set `CNI_CONF_NAME` to the filename for your k8s cluster's CNI config file in `/etc/cni/net.d`
+      1. set `exclude_namespaces` to include the namespace the Istio control-plane is installed in
+      1. set `cni_bin_dir` to your kubernetes install's CNI bin location (the value of kubelet's `--cni-bin-dir`)
+         1. default is `/opt/cni/bin`
```
Comment (venilnoronha) [inline README.md]: nit: set -> Set, default -> Default

File: README.md [hunk 20]
```diff
@@ -143,13 +173,55 @@ $ GOOS=linux make docker.push
...
+   1. The repo URL will be output (`http://127.0.0.1:8879`)
+
+To use this repo via `helm install`:
+
+```sh
+$ helm repo add local_istio http://127.0.0.1:8879
+$ helm repo update
+```
+
+
```
Comment (venilnoronha) [inline README.md]: nit: extra line feed

File: README.md [hunk 21]
```diff
@@ -90,7 +120,7 @@ of hosted Kubernetes environments and whether `istio-cni` has been trialed in th
 
 1. Install Istio
 
-1. remove the `initContainers` section from the result of helm template's rendering of
+1. remove the `initContainers` section from the result of Helm template's rendering of
```
Comment (venilnoronha) [inline README.md]: nit: remove -> Remove

File: README.md (hunk 19 above)
Comment (tiswanso) [reply README.md]: done

File: README.md (hunk 21 above)
Comment (tiswanso) [reply README.md]: done

File: README.md (hunk 20 above)
Comment (tiswanso) [reply README.md]: done

**Comment by venilnoronha** [comment]
@tiswanso thanks for fixing those!

**Pull request merged**